src/main/java/gyro/google/kms/CryptoKeyResource.java

```diff
@@ -0,0 +1,382 @@
+package gyro.google.kms;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.google.cloud.kms.v1.CryptoKey;
+import com.google.cloud.kms.v1.CryptoKeyPathName;
+import com.google.cloud.kms.v1.KeyManagementServiceClient;
+import com.google.cloud.kms.v1.KeyRingName;
+import com.google.protobuf.Duration;
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.Timestamp;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Min;
+import gyro.core.validation.Regex;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidationError;
+import gyro.google.Copyable;
+import gyro.google.GoogleResource;
+
+/**
+ * Add a cryto key to a king ring.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    google::crypto-key crypto-key-example
+ *        key-ring: $(google::key-ring example-key-ring)
+ *        name: "crypto-key-ring-example"
+ *        purpose: ENCRYPT_DECRYPT
+ *        rotation-period: 1
+ *        next-rotation-date: "09/21/2020"
+ *        primary-key-version-id: 1
+ *
+ *        crypto-key-version-template
+ *            algorithm: GOOGLE_SYMMETRIC_ENCRYPTION
+ *            protection-level: SOFTWARE
+ *        end
+ *    end
+ */
+@Type("crypto-key")
+public class CryptoKeyResource extends GoogleResource implements Copyable<CryptoKey> {
+
+    private KeyRingResource keyRing;
+    private String name;
+    private Long rotationPeriod;
+    private CryptoKey.CryptoKeyPurpose purpose;
```

Comment:
If `import com.google.cloud.kms.v1.CryptoKey;` is included
```suggestion
    private CryptoKeyPurpose purpose;
```